### ISSUE TITLE
Histogram timestamp

### ISSUE THREAD

**Issue by rohanganpatye** [lucidworks/banana#38]
Hi, 

Why does the histogram show a default timestamp of 17:00:00?? 

My query explicitly specifies the date and time to be 00:00:00 as I'm doing a histogram of daily events. While the count and date appears correct, each event shows the same timestamp of 17:00:00.
Can anyone tell me if I'm missing something here?

Thanks!

![image](https://cloud.githubusercontent.com/assets/2896203/4222228/66ba6f80-3911-11e4-92b3-b22d0fbb39e1.png)

**Comment by banajit** [comment]
Banana is converting solr UTC value to your browser time in tooltip.
You need to change following code, and in configuration change the mode to utc
$tooltip
.html(
group + dashboard.numberWithCommas(value) + " @ " + moment(item.datapoint[0]).format('MM/DD HH:mm:ss')
)
.place_tt(pos.pageX, pos.pageY);

Replace above code with below loc

if(scope.panel.timezone === 'utc') {
    var tooltipTime = moment.utc(item.datapoint[0]).format('MM/DD HH:mm:ss');
}
else {
    var tooltipTime = moment(item.datapoint[0]).format('MM/DD HH:mm:ss');
}
$tooltip
.html(
group + dashboard.numberWithCommas(value) + " @ " + tooltipTime
)

**Comment by rohanganpatye** [comment]
Hi, 

Thanks for your help, but I'm still just getting started with Banana. Can you please tell me which file or where exactly I need to make this change??

 Also, when you say configuration, do you mean on the histogram panel in the browser?
Thanks.

**Comment by banajit** [comment]
Yes, configuration means, configure panel in the browser.
You need to make change in module.js file, location 
banana\src\app\panels\histogram

**Comment by rohanganpatye** [comment]
Banajit, 

I tried doing what you recommended, but I see no change.

**Comment by banajit** [comment]
Have you change the mode to utc, in the configuration?
Also don't forget to clear the browser cache.

**Comment by rohanganpatye** [comment]
Yes, I cleared the browser cache and made the changes . However I do not see anything now when I hover over the histrogram. 

Also none of the points plotted on the histogram are selectable to be added as a filter to the query. I have configured them from the panel to be selectable.

Here is my module.js code:

var $tooltip = $('<div>');
        elem.bind("plothover", function (event, pos, item) {
          var group, value;
          if (item) {
            if (item.series.info.alias || scope.panel.tooltip.query_as_alias) {
              group = '<small style="font-size:0.9em;">' +
                '<i class="icon-circle" style="color:'+item.series.color+';"></i>' + ' ' +
                (item.series.info.alias || item.series.info.query)+
              '</small><br>';
            } else {
              group = kbn.query_color_dot(item.series.color, 15) + ' ';
            }
            if (scope.panel.stack && scope.panel.tooltip.value_type === 'individual')  {
              value = item.datapoint[1] - item.datapoint[2];
            } else {
              value = item.datapoint[1];
            }
                if(scope.panel.timezone === 'utc') {
                        var tooltipTime = moment.utc(item.datapoint[0]).format('MM/DD HH:mm:ss');
                }
                else {
                        var tooltipTime = moment(item.datapoint[0]).format('MM/DD HH:mm:ss');
                }
        $tooltip
          .html(
                group + dashboard.numberWithCommas(value) + " @ " + tooltipTime
                )
          } else {
            $tooltip.detach();
          }
        });

**Comment by banajit** [comment]
Seems there is no issue in the code. Please check the browser console whether you can see any error message there

**Comment by aadel** [comment]
@rohanganpatye could you post the query that you have in the "Inspect" dialog?

**Comment by rohanganpatye** [comment]
@aadel 
q=_:_&df=*&wt=json&rows=0&fq=DATE_TIME:[2014-05-20T00:00:00.000Z%20TO%202014-08-31T00:00:00.000Z]&facet=true&facet.range=DATE_TIME&facet.range.start=2014-05-20T00:00:00.000Z/DAY&facet.range.end=2014-08-31T00:00:00.000Z%2B1DAY/DAY&facet.range.gap=%2B1DAY

**Comment by aadel** [comment]
Please sync with the latest commit on develop branch 6d613927807802acb62620a6a124c35482df7c35 it has a fix related to this issue.

Please note that the time in TimePicker panel is local time and if Histogram time correction is browser, you should see 00:00:00 in the tooltip and UTC time in the query inspector. In fact, you should always see UTC time in query inspector as Solr stores date in ISO 8601 format only.
